dbms/src/Flash/Coprocessor/RemoteRequest.cpp

```diff
@@ -36,15 +42,8 @@ RemoteRequest RemoteRequest::build(const RegionRetryList & retry_regions, DAGCon
     DAGSchema schema;
     tipb::DAGRequest dag_req;
     auto * executor = dag_req.mutable_root_executor();
-    if (selection != nullptr)
-    {
-        executor->set_tp(tipb::ExecType::TypeSelection);
-        executor->set_executor_id(selection->executor_id());
-        auto * new_selection = executor->mutable_selection();
-        for (const auto & condition : selection->selection().conditions())
-            *new_selection->add_conditions() = condition;
-        executor = new_selection->mutable_child();
-    }
+
+    executor = push_down_filter.constructSelectionForRemoteRead(executor);
```

Comment:
```suggestion
    auto * executor = push_down_filter.constructSelectionForRemoteRead(dag_req.mutable_root_executor());
```
nit: maybe merge L44 and L46?